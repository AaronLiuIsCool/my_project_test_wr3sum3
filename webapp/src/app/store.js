import { configureStore } from '@reduxjs/toolkit';
import resumeReducer from 'features/SmartResume/slicers/resumeSlicer';
import counterReducer from 'features/CounterExample/counterSlice.js';

export default configureStore({
  reducer: {
    counter: counterReducer,
    resume: resumeReducer
  },
});
