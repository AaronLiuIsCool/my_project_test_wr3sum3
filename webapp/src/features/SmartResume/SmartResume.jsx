import React from 'react';

import LeftNav from './components/LeftNav';
import ExperiencesForm from './components/ExperiencesForm';
import Preview from './components/Preview';

const SmartResume = () => (
    <div>
        <div>智能创建</div>
        <LeftNav />
        <ExperiencesForm />
        <Preview />
    </div>
);

export default SmartResume;