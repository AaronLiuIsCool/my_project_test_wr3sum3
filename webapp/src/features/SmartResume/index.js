import React, { Suspense } from 'react';

import LoadFallbackComponent from 'components/LoadFallbackComponent';
import './styles/index.scss';

const SmartResume = React.lazy(() => import('./components/SmartResume'));

const LazyLoadSmartResume = ({ match }) => (
  <Suspense fallback={<LoadFallbackComponent />}>
    <SmartResume useObserver={true} resumeId={match.params.resumeId} />
  </Suspense>
);

export default LazyLoadSmartResume;
