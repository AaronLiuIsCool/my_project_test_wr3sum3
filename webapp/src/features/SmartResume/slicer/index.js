import { createSlice } from "@reduxjs/toolkit";

import assistant from './assistant';
import photoReference from "./photoReference";
import basic, {validateBasic} from "./basic";
import education, {validateEducation} from "./education";
import work, {validateWork} from "./work";
import project, {validateProject} from "./project";
import volunteer, {validateVolunteer} from "./volunteer";
import certificate, {validateCertificate} from "./certificate";
import ratings from "./ratings";
import resumeBuilder from "./resumeBuilder";
import { resumeAdaptor } from "../utils/slicerAdaptor";

export const resumeSlice = createSlice({
    name: "resume",
    initialState: {
        id: undefined,
        stepIndex: 0,

        assistant: assistant.initialState,
        photoReference: photoReference.initialState,
        basic: basic.initialState,
        education: education.initialState,
        work: work.initialState,
        project: project.initialState,
        certificate: certificate.initialState,
        volunteer: volunteer.initialState,
        ratings: ratings.initialState,
        resumeBuilder: resumeBuilder.initialState,
    },
    reducers: {
        setId: (state, action) => {
            state.id = action.payload;
        },

        setResume: (state, action) => {
            const resume = action.payload;
            return { ...state, ...resumeAdaptor(resume) };
        },

        moveStep: (state, action) => {
            state.stepIndex = action.payload;
        },

        ...assistant.reducers,
        ...photoReference.reducers,
        ...basic.reducers,
        ...education.reducers,
        ...work.reducers,
        ...project.reducers,
        ...volunteer.reducers,
        ...certificate.reducers,
        ...ratings.reducers,
        ...resumeBuilder.reducers,
    }
});

export const { actions } = resumeSlice;

export const selectId = ({ resume }) => resume.id;
export const selectStep = ({ resume }) => resume.stepIndex;
export const selectResume = ({ resume }) => resume;
export const selectValidatedResume = ({ resume }) => {
  const {photoReference, basic, education, work,
    project, volunteer, certificate} = resume;

  return {
    photoReference,
    basic: validateBasic(basic) ? basic : undefined,
    education: {...education, data: education.data.filter(validateEducation)},
    work: {...work, data: work.data.filter(validateWork)},
    project: {...project, data: project.data.filter(validateProject)},
    volunteer: {...volunteer, data: volunteer.data.filter(validateVolunteer)},
    certificate: {...certificate, data: certificate.data.filter(validateCertificate)},
  };
};

export const assistantSelectors = assistant.selectors;
export const photoReferenceSelectors = photoReference.selectors;
export const basicSelectors = basic.selectors;
export const educationSelectors = education.selectors;
export const workSelectors = work.selectors;
export const projectSelectors = project.selectors;
export const volunteerSelectors = volunteer.selectors;
export const certificateSelectors = certificate.selectors;
export const ratingsSelectors = ratings.selectors;
export const resumeBuilderSelectors = resumeBuilder.selectors;

export default resumeSlice.reducer;
