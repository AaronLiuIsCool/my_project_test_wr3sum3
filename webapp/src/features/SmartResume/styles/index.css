/* This file contains some shared css that can be used cross SmartResume */

.form_h1{
  text-align:left;
  margin: 40px 0;
}

.form_h1 h1{
  font-size: 30px;
  line-height: 1;
  color: #333333;
}

.form_h1 p{
  font-size: 14px;
  line-height: 2.14;
  color: #bfc4d3;
}

.form_h2{
  font-size: 20px;
  font-weight: bold;
  color: #333333;
}

.form_body{
  width: 800px;
  border-radius: 8px;
  box-shadow: 0 0 6px 0 rgba(0, 0, 0, 0.1);
  padding: 24px 20px;
}

.form_item label{
  color: #bfc4d3;
}
.form_item .rbt-input-main,
.form_item input{
  background-color: #f2f4f9;
  color: #d5d9e6;
  border-radius: 8px;
  border:none;
  /* box-shadow: 0 0 6px 0 rgba(0, 0, 0, 0.1); */
}
.space_betweens{
  display: flex;
  justify-content: space-between;
}
.form_reset_link{
  font-size: 16px;
  font-weight: bold;
  color: #d5d9e6;
}


/* 添加其他院校/工作/专业项目经历 按钮 */
.addMoreForm{
  display:flex;
  padding:20px;
  color: #3e89ec;
}
/* if svg is provided, we don't need this  */
.addMoreForm span{
  font-size: 50px;
  line-height: 20px;
}
.addMoreForm p{
  font-size: 20px;
  font-weight: bold;
  padding-left:10px;
}