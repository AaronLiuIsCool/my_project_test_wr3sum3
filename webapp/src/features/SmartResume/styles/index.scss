/* This file contains some shared css that can be used cross SmartResume */
.smart-resume {
  position: relative;

  a:hover {
    text-decoration: none;
  }

  .overlay {
    position: fixed;
    width: 100%;
  }

  .form_h1 {
    text-align: left;
    margin: 40px 0;

    div {
      display: flex;
      align-items: center;

      h1 {
        font-size: 30px;
        line-height: 1;
        color: #333333;
        margin-left: 10px;
        // icon has top margin, so we need manually adjust this 
        margin-top: 5px;
      }
    }

    p {
      font-size: 14px;
      line-height: 2.14;
      color: #bfc4d3;
    }
  }

  .form_h2 {
    font-size: 20px;
    font-weight: bold;
    color: #333333;
    margin-bottom: 10px;
  }

  .form_body {
    width: 800px;
    border-radius: 8px;
    box-shadow: 0 0 6px 0 rgba(0, 0, 0, 0.1);
    padding: 24px 20px;
  }

  .form_item {
    label {
      color: #bfc4d3;
    }

    .rbt-input-main,
    textarea,
    input {
      background-color: #f2f4f9;
      color: #333333;
      border-radius: 8px;
      border: none;
      box-shadow: 0 0 6px 0 rgba(0, 0, 0, 0.1);
    }

    .DateInput_input {
      height: 38px;
      font-size: 16px;
      line-height: 24px;

      &::placeholder {
        color: black;
      }
    }

    .radioButton_container {
      display: flex;
      justify-content: space-around;

      .radioButton {
        width: 38px;
        height: 38px;
        border-radius: 21px;
        border: none;
        background: white;
        color: #bfc4d3;

        &.selected {
          color: white;
          background-color: #3e89ec;
        }

        &:focus {
          border: none;
          outline: none;
        }
      }
    }


  }

  .form-group textarea {
    background-color: #f2f4f9;
    color: #d5d9e6;
    border-radius: 8px;
    border: none;
    color: #333;
  }

  .form_item .rbt-input-main,
  .form_item input {
    background-color: #f2f4f9;
    color: #d5d9e6;
    border-radius: 8px;
    border: none;
    color: #333;
  }

  .space_betweens {
    display: flex;
    justify-content: space-between;
  }

  .form_reset_link {
    font-size: 16px;
    font-weight: bold;
    color: #d5d9e6;
  }

  .w-100 {
    width: 100%;
    border-radius: 30px;
  }

  .SingleDatePicker {
    display: block;

    .SingleDatePickerInput {
      width: 100%;

      .DateInput {
        width: 100%;
      }
    }
  }


  /* 添加其他院校/工作/专业项目经历 按钮 */
  .addMoreForm {
    display: flex;
    padding: 20px;
    color: #3e89ec;
    cursor:pointer;

    /* if svg is provided, we don't need this  */
    span {
      font-size: 50px;
      line-height: 20px;
    }

    p {
      font-size: 20px;
      font-weight: bold;
      padding-left: 10px;
    }
  }

  .writeAssistantContainer {
    position: relative;

    .writeAssistant {
      position: absolute;
      top: 0;
      right: 0;
      cursor: pointer;

      .hand {
        fill: #bfc4d3;
      }

      .btn.btn-link {
        padding-left: 10px;
        font-size: 14px;
        font-weight: bold;
        color: #bfc4d3;
      }
    }

    &.active, &:hover {
      .hand {
        fill: #3e89ec;
      }

      .btn.btn-link {
        text-decoration: none;
        color: #3e89ec;
      }
    }
  }

  // input placeholder text color 
  ::-webkit-input-placeholder {
    /* Chrome/Opera/Safari */
    color: #d5d9e6 !important;
  }

  ::-moz-placeholder {
    /* Firefox 19+ */
    color: #d5d9e6 !important;
  }

  :-ms-input-placeholder {
    /* IE 10+ */
    color: #d5d9e6 !important;
  }

  :-moz-placeholder {
    /* Firefox 18- */
    color: #d5d9e6 !important;
  }

  // overwrite some date picker style 
  .CalendarDay__selected,
  .CalendarDay__selected:active,
  .CalendarDay__selected:hover {
    background: #3e89ec !important;
    border-color: #3e89ec !important;
  }

  .DayPickerKeyboardShortcuts_show__bottomRight::before {
    border-top: 26px solid transparent !important;
    border-right: 33px solid #3e89ec !important;
  }

  /* resume preview style  */
  .react-pdf__Page canvas{
    margin:20px auto;
  }
}