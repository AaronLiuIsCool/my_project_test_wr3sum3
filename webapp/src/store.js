import { configureStore } from '@reduxjs/toolkit';

import appReducer from 'features/App/slicer';
import accountReducer from 'features/App/slicer/account';
import resumeReducer from 'features/SmartResume/slicer';
import jobsMatcherReducer from 'features/JobsMatcher/slicer';

export default configureStore({
  reducer: {
    app: appReducer,
    account: accountReducer,
    resume: resumeReducer,
    jobs: jobsMatcherReducer,
  },
});
