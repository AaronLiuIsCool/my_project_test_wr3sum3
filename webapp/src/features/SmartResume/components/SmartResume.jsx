import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useSelector, useDispatch } from 'react-redux';
import { previewResume } from './ResumePreview/resumeBuilder';

import { selectLanguage, selectUserId } from 'features/App/slicer';
import { actions } from 'features/SmartResume/slicer';
import { I8nContext } from 'shell/i18n';
import AccountServices from 'shell/services/AccountServices';
import ResumeServices from 'shell/services/ResumeServices';
import { getLogger } from 'shell/logger';

import LeftNav from './LeftNav';
import ExperiencesForm from './ExperiencesForm';
import Assistant from './Assistant';
import ResumePreview from './ResumePreview';

import zh from '../i18n/zh.json';
import en from '../i18n/en.json';

const logger = getLogger('ResumeStarter');
const accountServices = new AccountServices();
const resumeServices = new ResumeServices();

async function getOrCreateResume(dispatch, userId, resumeId, language) {
    if (resumeId) {
        try {
            const response = await resumeServices.getResume(resumeId);
            const resumeData = await response.json();
            dispatch(actions.setResume(resumeData));
        } catch (exception) {
            logger.error(exception)
        }
        return;
    }

    try {
        const response = await resumeServices.createResume({language});
        const data = await response.json();
        const resumeId = data.id;
        accountServices.addResume(userId, resumeId);
        dispatch(actions.setId(resumeId));
    } catch (exception) {
        logger.error(exception);
        // TODO: Need to handle retry
    }
}

const SmartResume = ({ useObserver = false, resumeId }) => {
    const dispatch = useDispatch();
    const userId = useSelector(selectUserId);
    const language = useSelector(selectLanguage);
    const messages = language === 'zh' ? zh : en;

    useEffect(() => {
        getOrCreateResume(dispatch, userId, resumeId, language);
		previewResume(messages.RPreview);
    }, []); // eslint-disable-line

    return (
        <I8nContext.Provider value={messages}>
            <div className="features smart-resume">
                <div className="overlay">
                    <LeftNav />
                    <Assistant />
                    <ResumePreview  />
                </div>
                <ExperiencesForm useObserver={useObserver} />
            </div>
        </I8nContext.Provider>
    );
};

SmartResume.propTypes = {
    useObserver: PropTypes.bool 
};

export default SmartResume;