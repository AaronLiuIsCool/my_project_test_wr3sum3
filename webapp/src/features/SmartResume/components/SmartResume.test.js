import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';

import store from 'app/store';
import SmartResume from './SmartResume';

describe('SmartResume tests', () => {
  test('renders Smart Resume', () => {
    const smartResumeApp = renderer.create(
      <Provider store={store}>
        <SmartResume />
      </Provider>
    ).toJSON();

    expect(smartResumeApp).toMatchSnapshot();
  });

});
