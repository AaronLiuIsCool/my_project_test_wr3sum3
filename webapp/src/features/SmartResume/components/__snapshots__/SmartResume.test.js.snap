// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmartResume tests renders Smart Resume 1`] = `
<div
  className="features smart-resume"
>
  <div
    className="overlay"
  >
    <div
      className="container"
    >
      <div
        className="content"
      >
        <a
          className="navItemSelected"
          href="/#basicInfo"
          onClick={[Function]}
        >
          基本信息
        </a>
        <a
          className="navItem"
          href="/#education"
          onClick={[Function]}
        >
          教育背景
        </a>
        <a
          className="navItem"
          href="/#workXp"
          onClick={[Function]}
        >
          工作经历
        </a>
        <a
          className="navItem"
          href="/#projectXp"
          onClick={[Function]}
        >
          项目经历
        </a>
        <a
          className="navItem"
          href="/#otherXp"
          onClick={[Function]}
        >
          其他经历
        </a>
        <a
          className="navItem"
          href="/#certifications"
          onClick={[Function]}
        >
          技能证书
        </a>
      </div>
    </div>
    <div
      className="container"
    >
      TODO: Preview on right
    </div>
  </div>
  <div
    className="container"
  >
    <p>
      TODO: update progress bar
    </p>
    <div
      id="basicInfo"
    >
      <div
        className="form_h1"
      >
        <div>
          <h1>
            基本信息
          </h1>
        </div>
        <p>
          请输入您的一些基本个人信息
        </p>
      </div>
      <div
        className="form_body"
      >
        <div
          className="container"
        >
          <form
            className=""
          >
            <div
              className="row"
            />
            <div
              className="row"
            >
              <div
                className="col"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="basic-name-cn"
                  >
                    中文姓名
                  </label>
                  <input
                    className="form-control"
                    id="basic-name-cn"
                    placeholder="请输入您的中文全名"
                    type="text"
                  />
                </div>
              </div>
              <div
                className="col"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="basic-name-en"
                  >
                    英文称呼
                  </label>
                  <input
                    className="form-control"
                    id="basic-name-en"
                    placeholder="First & Last Name"
                    type="text"
                  />
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col-lg-6"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="basic-email"
                  >
                    邮箱
                  </label>
                  <input
                    className="form-control"
                    id="basic-email"
                    placeholder="请填写您的邮箱地址"
                    type="email"
                  />
                </div>
              </div>
              <div
                className="col-lg-3"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="basic-phone"
                  >
                    联系电话
                  </label>
                  <input
                    className="form-control"
                    id="basic-phone"
                    placeholder="输入您的联系电话"
                    type="text"
                  />
                </div>
              </div>
              <div
                className="col-lg-3"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="basic-city"
                  >
                    城市
                  </label>
                  <div
                    className="rbt"
                    style={
                      Object {
                        "outline": "none",
                        "position": "relative",
                      }
                    }
                    tabIndex={-1}
                  >
                    <div
                      style={
                        Object {
                          "display": "flex",
                          "flex": 1,
                          "height": "100%",
                          "position": "relative",
                        }
                      }
                    >
                      <input
                        aria-autocomplete="both"
                        aria-expanded={false}
                        aria-haspopup="listbox"
                        autoComplete="off"
                        className="rbt-input-main form-control rbt-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="您现在的居住城市"
                        role="combobox"
                        type="text"
                        value=""
                      />
                      <input
                        aria-hidden={true}
                        className="rbt-input-hint"
                        readOnly={true}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "borderColor": "transparent",
                            "boxShadow": "none",
                            "color": "rgba(0, 0, 0, 0.35)",
                            "left": 0,
                            "pointerEvents": "none",
                            "position": "absolute",
                            "top": 0,
                            "width": "100%",
                          }
                        }
                        tabIndex={-1}
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="col"
              />
              <div
                className="col"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="col"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="basic-linkedin"
                  >
                    个人领英 Linkedin 链接
                  </label>
                  <input
                    className="form-control"
                    id="basic-linkedin"
                    placeholder="请填写您的Linkedin链接"
                    type="text"
                  />
                </div>
              </div>
              <div
                className="col"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="basic-weblink"
                  >
                    个人 Github 或者作品集链接
                  </label>
                  <input
                    className="form-control"
                    id="basic-weblink"
                    placeholder="请填写您的Github链接或作品集链接"
                    type="text"
                  />
                </div>
              </div>
            </div>
            <div
              className="form_buttons row"
            >
              <div
                className="space_betweens col"
              >
                <button
                  className="form_reset_link btn btn-link"
                  disabled={false}
                  type="button"
                >
                  清空重填
                </button>
                <button
                  className="btn btn-primary"
                  disabled={false}
                  type="submit"
                >
                  保存
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
    <div
      id="education"
    >
      <div
        className="form_h1"
      >
        <div>
          <h1>
            教育背景
          </h1>
        </div>
        <p>
          请添加关于您教育背景的相关信息
        </p>
      </div>
      <div
        className="form_body"
      >
        <div
          className="container"
        >
          <form
            className=""
          >
            <div
              className="row"
            >
              <div
                className="col"
              >
                <h2
                  class="form_h2"
                >
                  填写院校信息
                </h2>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col-lg-4"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="education-school"
                  >
                    院校名称
                  </label>
                  <div
                    className="rbt"
                    style={
                      Object {
                        "outline": "none",
                        "position": "relative",
                      }
                    }
                    tabIndex={-1}
                  >
                    <div
                      style={
                        Object {
                          "display": "flex",
                          "flex": 1,
                          "height": "100%",
                          "position": "relative",
                        }
                      }
                    >
                      <input
                        aria-autocomplete="both"
                        aria-expanded={false}
                        aria-haspopup="listbox"
                        autoComplete="off"
                        className="rbt-input-main form-control rbt-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="请输入学校名字"
                        role="combobox"
                        type="text"
                        value=""
                      />
                      <input
                        aria-hidden={true}
                        className="rbt-input-hint"
                        readOnly={true}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "borderColor": "transparent",
                            "boxShadow": "none",
                            "color": "rgba(0, 0, 0, 0.35)",
                            "left": 0,
                            "pointerEvents": "none",
                            "position": "absolute",
                            "top": 0,
                            "width": "100%",
                          }
                        }
                        tabIndex={-1}
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="col-lg-2"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="education-gpa"
                  >
                    毕业GPA
                  </label>
                  <input
                    className="form-control"
                    id="education-gpa"
                    placeholder="4.0?"
                    type="text"
                  />
                </div>
              </div>
              <div
                className="col"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="education-enter-date"
                  >
                    入学日期
                  </label>
                  <input
                    className="form-control"
                    id="education-enter-date"
                    placeholder="年/月/日期"
                    type="text"
                  />
                </div>
              </div>
              <div
                className="col"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="education-graduate-date"
                  >
                    毕业日期
                  </label>
                  <input
                    className="form-control"
                    id="education-graduate-date"
                    placeholder="年/月/日期"
                    type="text"
                  />
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="education-major"
                  >
                    专业
                  </label>
                  <div
                    className="rbt"
                    style={
                      Object {
                        "outline": "none",
                        "position": "relative",
                      }
                    }
                    tabIndex={-1}
                  >
                    <div
                      style={
                        Object {
                          "display": "flex",
                          "flex": 1,
                          "height": "100%",
                          "position": "relative",
                        }
                      }
                    >
                      <input
                        aria-autocomplete="both"
                        aria-expanded={false}
                        aria-haspopup="listbox"
                        autoComplete="off"
                        className="rbt-input-main form-control rbt-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="输入所在的专业"
                        role="combobox"
                        type="text"
                        value=""
                      />
                      <input
                        aria-hidden={true}
                        className="rbt-input-hint"
                        readOnly={true}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "borderColor": "transparent",
                            "boxShadow": "none",
                            "color": "rgba(0, 0, 0, 0.35)",
                            "left": 0,
                            "pointerEvents": "none",
                            "position": "absolute",
                            "top": 0,
                            "width": "100%",
                          }
                        }
                        tabIndex={-1}
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="col"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="education-degree"
                  >
                    学历
                  </label>
                  <div
                    className="rbt"
                    style={
                      Object {
                        "outline": "none",
                        "position": "relative",
                      }
                    }
                    tabIndex={-1}
                  >
                    <div
                      style={
                        Object {
                          "display": "flex",
                          "flex": 1,
                          "height": "100%",
                          "position": "relative",
                        }
                      }
                    >
                      <input
                        aria-autocomplete="both"
                        aria-expanded={false}
                        aria-haspopup="listbox"
                        autoComplete="off"
                        className="rbt-input-main form-control rbt-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="请输入您获得的学历"
                        role="combobox"
                        type="text"
                        value=""
                      />
                      <input
                        aria-hidden={true}
                        className="rbt-input-hint"
                        readOnly={true}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "borderColor": "transparent",
                            "boxShadow": "none",
                            "color": "rgba(0, 0, 0, 0.35)",
                            "left": 0,
                            "pointerEvents": "none",
                            "position": "absolute",
                            "top": 0,
                            "width": "100%",
                          }
                        }
                        tabIndex={-1}
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="col"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="education-school-city"
                  >
                    学院所在城市
                  </label>
                  <div
                    className="rbt"
                    style={
                      Object {
                        "outline": "none",
                        "position": "relative",
                      }
                    }
                    tabIndex={-1}
                  >
                    <div
                      style={
                        Object {
                          "display": "flex",
                          "flex": 1,
                          "height": "100%",
                          "position": "relative",
                        }
                      }
                    >
                      <input
                        aria-autocomplete="both"
                        aria-expanded={false}
                        aria-haspopup="listbox"
                        autoComplete="off"
                        className="rbt-input-main form-control rbt-input"
                        onBlur={[Function]}
                        onChange={[Function]}
                        onClick={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder="学院所在城市"
                        role="combobox"
                        type="text"
                        value=""
                      />
                      <input
                        aria-hidden={true}
                        className="rbt-input-hint"
                        readOnly={true}
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "borderColor": "transparent",
                            "boxShadow": "none",
                            "color": "rgba(0, 0, 0, 0.35)",
                            "left": 0,
                            "pointerEvents": "none",
                            "position": "absolute",
                            "top": 0,
                            "width": "100%",
                          }
                        }
                        tabIndex={-1}
                        value=""
                      />
                    </div>
                  </div>
                </div>
              </div>
              <div
                className="col"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="education-school-country"
                  >
                    学校所在国家
                  </label>
                  <input
                    className="form-control"
                    id="education-school-country"
                    placeholder="学校所在国家"
                    type="text"
                  />
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="education-highest-award"
                  >
                    所获最高奖项 (有任何奖项都是加分好事！)
                  </label>
                  <input
                    className="form-control"
                    id="education-highest-award"
                    placeholder="请填写您在校期间所获得的最高奖项"
                    type="text"
                  />
                </div>
              </div>
              <div
                className="col"
              >
                <div
                  className="form_item form-group"
                >
                  <label
                    className="form-label"
                    htmlFor="education-other-award"
                  >
                    其他获得奖项
                  </label>
                  <input
                    className="form-control"
                    id="education-other-award"
                    placeholder="请填写您在校期间所获得的其他奖项"
                    type="text"
                  />
                </div>
              </div>
            </div>
            <div
              className="form_buttons row"
            >
              <div
                className="space_betweens col"
              >
                <p
                  className="hidden"
                />
                <button
                  className="btn btn-primary"
                  disabled={false}
                  type="submit"
                >
                  保存
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
      <div
        className="addMoreForm"
      >
        <span>
          +
        </span>
        <p>
          添加新经历
        </p>
      </div>
    </div>
  </div>
</div>
`;
